### PR TITLE
fix marsjob env error and downgrade controller-gen to v0.3.0

### PR DESCRIPTION
**Pull request by SimonCqk**
1. fix env error occurred in mars job scheduler.
2. downgrade controller-gen to v0.3.0 since generated manifests files can be backward compatible.

### CODE REVIEW

File: config/crd/bases/kubedl.io_marsjobs.yaml [hunk 1]
```diff
@@ -1,3777 +1,165 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (jian-he) [inline config/crd/bases/kubedl.io_marsjobs.yaml:3]: 为什么这个是v1beta1

File: config/crd/bases/kubedl.io_marsjobs.yaml [hunk 2]
```diff
@@ -1,3777 +1,165 @@
...
-                                    type: string
-                                  whenUnsatisfiable:
-                                    description: 'WhenUnsatisfiable indicates how to deal with a pod if it doesn''t satisfy the spread constraint. - DoNotSchedule (default) tells the scheduler not to schedule it - ScheduleAnyway tells the scheduler to still schedule it It''s considered as "Unsatisfiable" if and only if placing incoming pod on any topology violates "MaxSkew". For example, in a 3-zone cluster, MaxSkew is set to 1, and pods with the same labelSelector spread as 3/1/1: | zone1 | zone2 | zone3 | | P P P |   P   |   P   | If WhenUnsatisfiable is set to DoNotSchedule, incoming pod can only be scheduled to zone2(zone3) to become 3/2/1(3/1/2) as ActualSkew(2-1) on zone2(zone3) satisfies MaxSkew(1). In other words, the cluster can still be imbalanced, but scheduler won''t make it *more* imbalanced. It''s a required field.'
-                                    type: string
-                                required:
-                                - maxSkew
-                                - topologyKey
-
```
Comment (jian-he) [inline config/crd/bases/kubedl.io_marsjobs.yaml:160]: 而这里是v1alpha1

File: config/crd/bases/kubeflow.org_pytorchjobs.yaml [hunk 3]
```diff
@@ -1,3749 +1,137 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (jian-he) [inline config/crd/bases/kubeflow.org_pytorchjobs.yaml:3]: why v1beta1 ?

File: config/crd/bases/kubeflow.org_pytorchjobs.yaml [hunk 4]
```diff
@@ -1,3749 +1,137 @@
...
-                                    type: string
-                                  whenUnsatisfiable:
-                                    description: 'WhenUnsatisfiable indicates how to deal with a pod if it doesn''t satisfy the spread constraint. - DoNotSchedule (default) tells the scheduler not to schedule it - ScheduleAnyway tells the scheduler to still schedule it It''s considered as "Unsatisfiable" if and only if placing incoming pod on any topology violates "MaxSkew". For example, in a 3-zone cluster, MaxSkew is set to 1, and pods with the same labelSelector spread as 3/1/1: | zone1 | zone2 | zone3 | | P P P |   P   |   P   | If WhenUnsatisfiable is set to DoNotSchedule, incoming pod can only be scheduled to zone2(zone3) to become 3/2/1(3/1/2) as ActualSkew(2-1) on zone2(zone3) satisfies MaxSkew(1). In other words, the cluster can still be imbalanced, but scheduler won''t make it *more* imbalanced. It''s a required field.'
-                                    type: string
-                                required:
-                                - maxSk
```
Comment (jian-he) [inline config/crd/bases/kubeflow.org_pytorchjobs.yaml:130]: and here v1

File: config/crd/bases/kubedl.io_marsjobs.yaml (hunk 1 above)
Comment (SimonCqk) [reply config/crd/bases/kubedl.io_marsjobs.yaml:3]: the `v1beta1` is the api version of `CustomizedResourceDefinition` resource in k8s 1.16.9 (controller-tool v0.3.0 depends on k8s 1.16.9 but controller-tool v0.4.0 depends on k8s v1.18.5), api version `v1`` of `CRD` resource was released by k8s 1.18

File: config/crd/bases/kubedl.io_marsjobs.yaml (hunk 2 above)
Comment (SimonCqk) [reply config/crd/bases/kubedl.io_marsjobs.yaml:160]: the v1alpha1 is the api version of `MarsJob`

**Review by tzzcfrank — APPROVED** [review]
LGTM

**Pull request merged**